src/Controls/src/Core/Registrar.cs

```diff
@@ -79,6 +82,18 @@ public void Register(Type tview, Type trender, Type[] supportedVisuals, short pr
 		/// <include file="../../docs/Microsoft.Maui.Controls.Internals/Registrar.xml" path="//Member[@MemberName='Register']/Docs" />
 		public void Register(Type tview, Type trender) => Register(tview, trender, _defaultVisualRenderers);
 
+		HashSet<Type> _registeredVisuals = new HashSet<Type>();
+
+		internal void RegisterVisual([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type registeredVisual)
+		{
+			if (!_registeredVisuals.Contains(registeredVisual))
+				_registeredVisuals.Add(registeredVisual);
```

Comment:
Since `_registeredVisuals` is a hashset, there is no need to check contains first. You can just add to it, and if it is already in there, Add returns `false`.